File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/selection/NotificationMessage.java
package rectangledbmi.com.pittsburghrealtimetracker.selection;

/**
 * Class instance to pass an object to create toasts and snackbar messages reactively.
 *
 * @author Jeremy Jao
 * @since 70
 */
public class NotificationMessage {

    public static NotificationMessage create(String message, int length) {
        return new NotificationMessage(message, length);
    }

    private String message;
    private int length;

    private NotificationMessage(String message, int length) {
        setMessage(message);
        setLength(length);
    }

    public String getMessage() {
        return message;
    }

    private void setMessage(String message) {
        this.message = message;
    }

    public int getLength() {
        return length;
    }

    private void setLength(int length) {
        this.length = length;
    }
}

File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/selection/NotificationMessage.kt
package rectangledbmi.com.pittsburghrealtimetracker.selection

/**
 * Class instance to pass an object to create toasts and snackbar messages reactively.
 *
 * @author Jeremy Jao
 * @since 70
 */
class NotificationMessage private constructor(message: String, length: Int) {

    var message: String? = null
        private set
    var length: Int = 0
        private set

    init {
        this.message = message
        this.length = length
    }

    companion object {

        @JvmStatic
        fun create(message: String, length: Int): NotificationMessage {
            return NotificationMessage(message, length)
        }
    }
}
